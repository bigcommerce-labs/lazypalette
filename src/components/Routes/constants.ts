export const enum NoMatch {
    Heading = '404 Error - Page not found',
    Body = 'Uh oh, looks like the page you are looking for has moved or no longer exists.',
}
