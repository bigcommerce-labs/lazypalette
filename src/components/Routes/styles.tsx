import styled from 'styled-components';

export const Container = styled.div`
    background: ${({ theme }) => theme.colors.empty};
    border-radius: 0px;
    box-shadow: ${({ theme }) => theme.elevation.raised};
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin: 0.75rem 0.75rem 0 0;
    height: calc(100% - 6rem);
    width: calc(100% - 15.5rem);
`;

export const Heading = styled.h1`
    color: ${({ theme }) => theme.colors.primaryText};
`;

export const Body = styled.p`
    color: ${({ theme }) => theme.colors.primaryText};
    font-size: ${({ theme }) => theme.typography.fontSize.small};
`;
