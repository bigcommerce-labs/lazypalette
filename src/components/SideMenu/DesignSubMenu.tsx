import React, { Component } from 'react';

import ButtonInput from '../ButtonInput/ButtonInput';

import SubMenu from './SubMenu';

interface MenuItem {
    label: string;
    path: string;
}

interface DesignSubMenuProps {
    sections: string[];
    currentPath: string;
    handleSave?(): void;
}

const staticItems: MenuItem[] = [
    {
        label: 'Logo and name',
        path: 'logo',
    },
    {
        label: 'Store theme',
        path: 'theme',
    },
];

const getItems = (sections: string[]) => {
    return sections.map((section: string, index: number) => (
        {
            label: section,
            path: `style/${index}`,
        }
    ));
};

class DesignSubMenu extends Component<DesignSubMenuProps> {
    handleSave = () => {
        this.props.handleSave!();
    };

    render() {
        return (
            <>
                <SubMenu
                    title="Design"
                    items={[...staticItems, ...getItems(this.props.sections)]}
                    currentPath={this.props.currentPath}
                    showArrows={false}
                />
                <ButtonInput
                    onClick={this.handleSave}
                    disabled={false}
                    classType="primary"
                    type="button"
                >
                    Save
                </ButtonInput>
            </>
        );
    }
}

export default DesignSubMenu;
