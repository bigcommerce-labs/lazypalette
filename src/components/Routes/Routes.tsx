import React, { PureComponent } from 'react';
import { withRouter, Route, RouteComponentProps, Switch } from 'react-router-dom';

import PreviewPane from '../PreviewPane/PreviewPane';

import { NoMatch } from './constants';
import { Body, Container, Heading } from './styles';

export const appRoutes = {
    history: {
        path: '/history',
        route: 'history',
    },
    home: {
        path: '/',
        route: '/',
    },
    options: {
        path: '/options',
        route: 'options',
    },
    section: {
        path: '/section/:id(\\d+)',
        route: 'section/',
    },
    styles: {
        path: '/styles',
        route: 'styles',
    },
};

export const NoRoute = () => (
    <Container>
        <Heading>{NoMatch.Heading}</Heading>
        <Body>{NoMatch.Body}</Body>
    </Container>
);

interface RoutesProps extends RouteComponentProps<{}> {}

export class Routes extends PureComponent<RoutesProps> {
    render() {
        const { history, home, options, styles, section  } = appRoutes;

        return (
            <Switch>
                <Route path={history.path} exact component={PreviewPane} />
                <Route path={home.path} exact component={PreviewPane} />
                <Route path={options.path} exact component={PreviewPane} />
                <Route path={styles.path} exact component={PreviewPane} />
                <Route path={section.path} exact component={PreviewPane} />
                <Route component={NoRoute} />
            </Switch>
        );
    }
}

export default withRouter(Routes);
