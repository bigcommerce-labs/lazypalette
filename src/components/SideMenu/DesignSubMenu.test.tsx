import { shallow } from 'enzyme';
import React from 'react';

import { appRoutes } from '../Routes/Routes';

import DesignSubMenu from './DesignSubMenu';

describe('DesignSubMenu', () => {
    const { styles } = appRoutes;

    it('renders', () => {
        const menu = shallow(
            <DesignSubMenu
                sections={['fred', 'joe']}
                currentPath={styles.path}
            />
        );

        expect(menu).toMatchSnapshot();
    });
});
