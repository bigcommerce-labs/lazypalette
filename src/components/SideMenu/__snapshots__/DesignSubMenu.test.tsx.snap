// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<React.Fragment>
  <SubMenu
    currentPath="/design/logo"
    items={
      Array [
        Object {
          "label": "Logo and name",
          "path": "logo",
        },
        Object {
          "label": "Store theme",
          "path": "theme",
        },
        Object {
          "label": "fred",
          "path": "style/0",
        },
        Object {
          "label": "joe",
          "path": "style/1",
        },
      ]
    }
    showArrows={false}
    title="Design"
  />
  <ButtonInput
    classType="primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Save
  </ButtonInput>
</React.Fragment>
`;
